src/main/java/pl/anasz/cinema/controllers/CinemaController.java

```diff
@@ -0,0 +1,191 @@
+        }
+        return seatResponse;
+    }
+
+    @PostMapping(value = "/summary",
+            produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
+    @ResponseBody
+    public ReservationResponse getReservationSummary(@RequestBody ReservationSummaryModel summaryModel) {
+        Reservation reservation = new Reservation();
+        ReservationResponse reservationResponse = new ReservationResponse();
+        Optional<Screening> screeningOptional = screeningService.findById(summaryModel.getScreeningId());
+        Screening screening = new Screening();
+        if (screeningOptional.isPresent()) {
+            screening = screeningOptional.get();
+        }
+
+        if(LocalDateTime.now().isAfter(screening.getScreeningTime().minusMinutes(15))){
+            reservationResponse.setValidationMessage("It's too late to reserve this screening");
+            return reservationResponse;
+        }
+
+        reservation.setMovieTitle(screening.getMovie().getTitle());
+        reservation.setExpirationDate(screening.getScreeningTime().minusMinutes(15));
+        reservation.setNumberOfTickets(summaryModel.getSeatsSignatures().size());
+
+        User booker = new User();
+        booker.setName(summaryModel.getBookerName());
+        booker.setSurname(summaryModel.getBookerSurname());
+
+        reservation.setBooker(booker);
+        reservation.setScreening(screening);
+
+        BigDecimal amount = calculateAmount(summaryModel.getTicketTypes());
+        reservation.setAmount(amount);
+
+        List<Seat> seatList = new ArrayList<>();
+        Optional<Long> roomIdOptional = screeningService.findRoomIdByScreeningId(summaryModel.getScreeningId());
+
+        summaryModel.getSeatsSignatures().forEach(seatSignature -> {
+            long roomId = -1;
+            if(roomIdOptional.isPresent()){
+                roomId = roomIdOptional.get();
+            }
+            Optional<Seat> seatOptional = seatsService.findSeatByRoomIdAndSignature(roomId, seatSignature);
+            if (seatOptional.isPresent()) {
+                Seat seat = seatOptional.get();
+                seat.setReservations(Collections.singletonList(reservation));
+                seatList.add(seat);
+            }
+
+        });
+
+        reservation.setSeats(seatList);
+        Optional<ReservationResponse> validation = validationService.validateReservation(reservation);
+        if(validation.isPresent()){
+            return validation.get();
+        }
+
+        userService.save(booker);
+        reservationsService.save(reservation);
+
+        reservationResponse.setAmount(amount);
+        reservationResponse.setReservationExpirationDate(screening.getScreeningTime().minusMinutes(15));
+        reservationResponse.setValidationMessage("SUCCESS!");
+
+        return reservationResponse;
+    }
+
+    private BigDecimal calculateAmount(List<String> ticketTypes) {
+        double adultTicket = 25;
```

Comment:
Zwracasz z tej metody BigDecimal, fajnie. Pytanie, czy dobrze inicjujesz i wykonujesz operacje na tych zmiennych? Czemu od razu nie trzymasz się tej klasy (BigDecimal), tylko zaczynasz od double?

Reply:
To prawda, dopóki ceny są proste. Warto jednak trzymać się jednego, spójnego podejścia - albo BigDecimal, albo double.